release-notes-1.2.0.md

```diff
@@ -0,0 +1,766 @@
+- Introduced new client connectors (email, slack and azure time-series).
+- Changed organization names of all supported connectors.
+- Support to mock functions when writing tests
+- Test and code coverage report
+- Java binding generator tool
+
+# What's new in JBallerina 1.2.0?
+
+## Language
+
+The language implementation is based on the stable language specification version 2020R1. This implementation introduces a new set of features aimed at making integration further easier for the user. In addition to that, the XML and Lock implementations are out in their preview/experimental status.
+
+In addition to new features, this release also includes critical bug fixes and fixes done to align the implementation with the language specification. Some of these changes are backward incompatible. For a complete list of changes done since 1.1.0, see the `Language changes since JBallerina 1.1.0` section.
+
+### New XML design
+
+This release introduces a revamped XML support along with XPath-like query syntax allowing easy and safe manipulation of XML data. 
+
+#### XML navigation expression
+
+XML step expressions allow to query the children of an XML element or children of members of an XML sequence.
+
+```ballerina
+xml x = xml `<root>
+<person><name><fname>Kamal</fname><lname>Gunae</lname></name></person>
+<person><name><fname>Nimal</fname><lname>Jayee</lname></name></person>
+<person><name><fname>Sunil</fname><lname>Kumarae</lname></name></person>
+</root>`;
+
+// Select all children items of `x`.
+xml allChildren = x/*;
+
+// Select all children items, which are XML elements.
+xml allElementChildren = x/<*>;
+
+// Match first-level children with element name `person`.
+// Then, match the second level with element name `name`
+// and the third level with element name `fname`.
+xml fnameItems = x/<person>/<name>/<fname>;
+
+// Select all descendants, which match element name `fname`.
+xml fnameDescendents = x/**/<fname>;
+```
+
+XML filter expression allows filtering an XML sequence by an element name.
+
+```ballerina
+xml x = xml `<root>
+<rectangle length="5" width="10"/>
+<rectangle length="5" width="5"/>
+<circle radius="2"/>
+<square side="6"/>
+</root>`;
+
+xml rectangles = x/*.<rectangle>;
+```
+
+#### XML attribute access
+
+ ```ballerina
+xmlns "www.ballerina.io/ns" as ns;
+xml val = xml `<element type="fixed" ns:count="2></element>`;
+string|error 'type = val.'type;
+string|error count = val.ns:count;
+```
+
+XML attribute access is now lax typed. This means that compile-time type checking is relaxed and moved to runtime. Accessing a non-existent attribute or field access expression on a non xml-element item will result in an error being returned. If optional field access syntax is used, then nil will be returned instead of error when the field is not available. Attributes with namespace prefix are accessed using the `value.prefix:attrName` syntax where the XML prefix must be declared in the scope.
```

Comment:
```suggestion
XML attribute access is now lax typed. This means that compile-time type checking is relaxed and moved to runtime. Accessing a non-existent attribute or using field access expression on a non-XML element item will result in an error being returned. If optional field access syntax is used, then nil will be returned instead of an error when the field is not available. Attributes with namespace prefix are accessed using the `value.prefix:attrName` syntax where the XML prefix must be declared in the scope.
```